### PR TITLE
dotCMS/core#12080 Add the new Vanity URL Content Types for the Add Co…

### PR DESCRIPTION
**Pull request by luisrojascr**
…ntent Types button

### CODE REVIEW

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts [hunk 1]
```diff
@@ -45,11 +45,16 @@ export class ToolbarAddContenletBodyComponent {
...
-    private types: StructureTypeView[];
+    types: StructureTypeView[];
+    mainTypes: StructureTypeView[];
+    moreTypes: StructureTypeView[];
+    moreItems = [];
+    private MAIN_CONTENT_TYPES = ['CONTENT', 'WIDGET', 'FORM', 'FILEASSET', 'HTMLPAGE'];
     private recent: StructureTypeView[];
-    private structureTypeViewSelected: StructureTypeView[];
-    private showMore = false;
+    structureTypeViewSelected: StructureTypeView[];
```
Comment (jdotcms) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: why are you removing the private?

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts [hunk 2]
```diff
@@ -65,24 +70,47 @@ export class ToolbarAddContenletComponent extends BaseComponent {
...
-                        this.recent.push(structure);
-                    } else {
-                        structure.types.forEach(type => {
-                            this.routingService.addPortletURL(type.name, type.action);
-                        });
+
+            this.recent = structures.filter(this.isRecentContentType);
+
+            this.types = structures.filter(structure => {
+                return !this.isRecentContentType(structure);
```
Comment (jdotcms) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: isn't the return redundant?

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts [hunk 3]
```diff
@@ -65,24 +70,47 @@ export class ToolbarAddContenletComponent extends BaseComponent {
...
-            );
+                });
+            });
 
             this.nextRecent();
         });
     }
 
+    private isRecentContentType(type: StructureTypeView): boolean {
+        return type.name.startsWith('RECENT');
```
Comment (jdotcms) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: shouldn't be a constant?

File: src/app/view/components/_common/action-button/action-button.component.html [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
-    icon="fa-plus" 
+    icon="fa-plus"
     class="action-button" 
     (onClick)="command()" 
     [model]="model" 
-    [class.multiple-options]="model && model.length">
-</p-splitButton>
\ No newline at end of file
+    [class.multiple-options]="model && model.length"
+    [class.toolbar-add-contentlet__head-button]="flat">
```
Comment (fmontes) [inline src/app/view/components/_common/action-button/action-button.component.html]: This class name should be descriptive like "flat" related.

File: src/app/view/components/_common/action-button/action-button.component.html [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
+</p-splitButton>
+<span *ngIf="label" class="showMore">{{ label }}</span>
```
Comment (fmontes) [inline src/app/view/components/_common/action-button/action-button.component.html]: This class name it's not BEM

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.html [hunk 6]
```diff
@@ -1,15 +1,17 @@
 <dot-dropdown-component
+    #dropdown
```
Comment (fmontes) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.html]: I think we don't need this, it was just for the test, right?

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss [hunk 7]
```diff
@@ -20,3 +20,77 @@
...
 }
 
+
+.toolbar-add-contentlet__head-button{
+    &.action-button{
+        background: none;
+        border: none;
+        box-shadow: none;
+        border-radius: 50%;
+        height: 32px;
```
Comment (fmontes) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss]: There is a var name for this.

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss [hunk 8]
```diff
@@ -20,3 +20,77 @@
...
+    
+        &:hover {
+            border: none;
+            box-shadow: none;
+            background: rgba(22, 22, 22, 0.1);
```
Comment (fmontes) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss]: There is a var for this.

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts (hunk 1 above)
Comment (fmontes) [reply src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: Private are not accessible from the template (yup weird) https://stackoverflow.com/questions/34574167/angular2-should-private-variables-be-accessible-in-the-template

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts (hunk 2 above)
Comment (fmontes) [reply src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: Arrow function only return when you do like:

```structures.filter(structure => !this.isRecentContentType(structure))```

But turns out that Typescript won't allow to do ```!``` like that.

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts (hunk 3 above)
Comment (fmontes) [reply src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: Maybe :)

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts [hunk 9]
```diff
@@ -65,24 +70,47 @@ export class ToolbarAddContenletComponent extends BaseComponent {
...
+                return !this.isRecentContentType(structure);
+            });
+
+            this.types.forEach(structure => {
+                structure.types.forEach(type => {
+                    this.routingService.addPortletURL(type.name, type.action);
+                });
+            });
+            this.mainTypes = this.getMainContentType(this.types);
+            this.moreTypes = this.getMoreContentTypes(this.types);
```
Comment (fmontes) [inline src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.ts]: We can use ```.map()``` to do this, let me know and I'll explain to you.

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.html (hunk 6 above)
Comment (luisrojascr) [reply src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.html]: Right, I'm removing that.

File: src/app/view/components/_common/action-button/action-button.component.html (hunk 5 above)
Comment (freddyucv) [reply src/app/view/components/_common/action-button/action-button.component.html]: Also I think showMore is not a good name, maybe something like showLabel is better.

File: src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss (hunk 7 above)
Comment (fmontes) [reply src/app/view/components/toolbar-add-contentlet/toolbar-add-contentlet.scss]: No there is not :)

**Pull request merged**